node/silkworm/downloader/internals/preverified_hashes.hpp

```diff
@@ -41,25 +41,23 @@ namespace silkworm {
  * preverified_hashes_mainnet.cpp
  *
  */
+
 struct PreverifiedHashes {
     std::set<evmc::bytes32> hashes;  // Set of hashes of headers that are known to belong to canonical chain
-    uint64_t height{0};              // Block height corresponding to the highest preverified header
+    uint64_t height{0};              // Block height corresponding to the highest pre-verified header
 
     [[nodiscard]] bool contains(const evmc::bytes32& hash) const { return hashes.find(hash) != hashes.end(); }
 
     static PreverifiedHashes none;  // A void set of hashes that can be used to turn-off pre-verified hashes usage and
                                     // that is useful for  (the default construction of) classes that use this
                                     // functionality optionally
 
-    static PreverifiedHashes mainnet;  // The mainnet set of pre-verified hashes
+    static const PreverifiedHashes& load(uint64_t chain_id); // Load a set of pre-verified hashes from low level impl
 
-    static std::map<uint64_t, const PreverifiedHashes&> per_chain;  // chain-id based access to the instances
+  private:
+    static std::map<uint64_t, PreverifiedHashes> per_chain;  // chain-id based access to the instances
```

Comment:
I don't get the point of this. Why a map is needed ?
Afaik there is no way we can use preverified hashes from two (or more) different chains

Reply:
On a wider perspective I don't get the point of the whole PreverifiedHashes struct.
Why not simply a pair<BlockNum,std::set<evmc::bytes32>> as const member of WorkingChain which is the only object using it ? Basically preverified hashes should be treated pretty much as ConsensusEngine: once is loaded it stays the same for all the duration of Silkworm instance.
Unless there are specific reasons I'd get rid of this struct

Reply:
The PreverifiedHashes struct has 2 fields as std::pair<>, yes, but the first makes the code clearer:
- `preverified_hashes_->height` is more clear than `preverified_hashes_.first`
- `preverified_hashes_->contains(...)` is more clear than `preverified_hashes_.second.find(...) ...`

Reply:
Regarding the map, I managed to remove it maintaining the possibility to use or not pre-verified hashes (e.g. with a command line switch).